pkg/cmd/roachtest/test_runner.go

```diff
@@ -892,6 +897,12 @@ func (r *testRunner) runWorker(
 				c = nil
 			}
 		}
+
+		// stop the tests if the failure rate has been exceeded
+		failureRate := float64(len(r.status.fail)) / float64(r.testCount)
```

Comment:
Grab the lock before reading `r.status.fail`?

Reply:
Yeah realised after pushing. Done.